strategy/thompson_sampling_strategy.py

```diff
@@ -0,0 +1,152 @@
+"""
+
+ThompsonSampling Strategy
+
+"""
+
+import numpy as np
+import scipy.stats
+from copy import deepcopy
+from strategy.strategy import Strategy
+
+
+class ThompsonSampling(Strategy):
+    """ Thompson Sampling Strategy
+
+    Args:
+        bandit (Bandit): bandit
+        **kwargs : prior parameters
+
+    Attributes:
+        num_arms (int)
+        posterior_params (list of list): posterior parameters
+        estimated_arm_means (ndarray): posterior predictive mean
+        estimated_arm_sds (ndarray): posterior predictive standard deviation
+    """
+    def __init__(self, bandit, **kwargs):
+        self.bandit = bandit
+        self.num_arms = bandit.num_arms
+        self.prior_params = [deepcopy(kwargs) for _ in range(self.num_arms)]
+        self.posterior_params = deepcopy(self.prior_params)
+
+    def fit(self, iterations, restart=False):
+        if restart == True:
+            self._restart()
+
+        index_arms_pulled = [None] * iterations
+        observed_rewards = [None] * iterations
+        estimated_arm_means = [None] * iterations
+        estimated_arm_sds = [None] * iterations
+        for i in range(iterations):
+            index_arms_pulled[i] = self._choose_arm()
+            observed_rewards[i] = self.pull_arm(index_arms_pulled[i])
+            self._update_posterior(index_arms_pulled[i], observed_rewards[i])
+            estimated_arm_means[i] = self.estimated_arm_means
+            estimated_arm_sds[i] = self.estimated_arm_sds
+
+        out = {
+            'rewards': np.array(observed_rewards),
+            'arms_pulled': np.array(index_arms_pulled),
+            'estimated_arm_means': np.array(estimated_arm_means),
+            'estimated_arm_sds': np.array(estimated_arm_sds)
+        }
+        return out
+
+    def _restart(self):
+        self.posterior_params = deepcopy(self.prior_params)
+
+    def _choose_arm(self):
+        samples = [self._sample(**params) for params in self.posterior_params]
+        return np.argmax(samples)
+
+    def pull_arm(self, arm_index):
```

Comment:
Just realized this isn't private.  Can you change that (or I can do it).  Users shouldn't be interacting with bandit directly except through .fit()